app/models/user.rb

```diff
@@ -24,4 +24,10 @@ def follows?(another_user)
   def can_follow?(another_user)
     !follows?(another_user) && self != another_user
   end
+
+  def generate_reset_link(scheme=nil, host=nil)
+    token = SecureRandom.urlsafe_base64
+    self.update(token: token)
+    "#{scheme}://#{host}/reset_password/#{self.token}"

```

Comment:
I don't think you should return the full URL here, just saving the token should be fine.